tmt/steps/report/reportportal.py

```diff
@@ -38,113 +25,179 @@ class ReportReportPortalData(tmt.steps.report.ReportStepData):
+        else:
+            self.debug(f"Response code from the server: {response.status_code}")
+            self.debug(f"Message from the server: {response.text}")
+
+    @staticmethod
+    def get_status(result: ResultOutcome) -> str:
+        if result == ResultOutcome.PASS:
+            return "PASSED"
+        elif result in [ResultOutcome.FAIL, ResultOutcome.ERROR, ResultOutcome.WARN]:
+            return "FAILED"
+        else:  # elif result == tmt.result.ResultOutcome.INFO:
+            return "SKIPPED"
+
     def go(self) -> None:
         """
-        Read executed tests, prepare junit, compress it to a zip zile and
-        send it to the ReportPortal instance.
+        Create a ReportPortal launch and its test items,
+        fill it with all parts needed and report the logs.
         """
 
         super().go()
 
-        # Check the data, show interesting info to the user
-        # Required fields are: url, token and project
         server = self.get("url")
         if not server:
             raise tmt.utils.ReportError("No ReportPortal server url provided.")
         server = server.rstrip("/")
 
+        project = self.get("project")
+        if not project:
+            raise tmt.utils.ReportError("No ReportPortal project provided.")
+
         token = self.get("token")
         if not token:
             raise tmt.utils.ReportError("No ReportPortal token provided.")
 
-        project = self.get("project")
-        if not project:
-            raise tmt.utils.ReportError("No ReportPortal project provided.")
-        self.info("project", project, color="green")
-
-        # Use provided launch name, default to run workdir name
-        assert self.step.plan.my_run is not None
-        assert self.step.plan.my_run.workdir is not None
-        launch_name = self.get("launch-name") or self.step.plan.my_run.workdir.name
-        self.info("launch", launch_name, color="green")
-
-        # Generate a xUnit report
-        import_junit_xml()
-        assert junit_xml is not None
-        suite = junit.make_junit_xml(self)
-        data = junit_xml.TestSuite.to_xml_string([suite])
-
-        # Zip the report
-        bytestream = io.BytesIO()
-        with zipfile.ZipFile(bytestream, "w", compression=zipfile.ZIP_DEFLATED,
-                             compresslevel=1) as zipstream:
-            # XML file names are irrelevant to ReportPortal
-            with zipstream.open("tests.xml", "w") as entry:
-                entry.write(data.encode("utf-8"))
-        bytestream.seek(0)
-
-        # Send the report to the ReportPortal instance
+        assert self.step.plan.name is not None
+        launch_name = self.get("launch") or self.step.plan.name
+
+        version = "v1"
+        url = f"{server}/api/{version}/{project}"
+        headers = {"Authorization": "bearer " + token,
+                   "accept": "*/*",
+                   "Content-Type": "application/json"}
+
+        TMT_env_vars = self.get("env-var") or []
```

Comment:
Also, with the change of the type I proposed above & `normalize=` addition, the call would always return a list, even if it would an empty one, so no need for `or []` anymore.

Reply:
Okay, fixed.